web/src/message_edit.ts

```diff
@@ -954,18 +953,7 @@ function do_toggle_resolve_topic(
     void channel.patch({
         url: "/json/messages/" + message_id,
         data: request,
-        success() {
-            if ($row) {
-                const $spinner = $row.find(".toggle_resolve_topic_spinner");
-                loading.destroy_indicator($spinner);
-            }
-        },
```

Comment:
What removes the loading indicator when it succeeds? Is it just the rerender of the message feed from the move having happened? We probably want a comment to that effect -- this code reads as wrong on an initial look, since it calls `show_toggle_resolve_topic_spinner` but never calls the hide function for it.

Reply:
Yes, it is the re-render of the message feed as you noted that removes the loading indicator on a successful response. Added a comment in the code for the same:
```
    // While we call the show_button_loading_indicator method to
    // show the spinner, we don't need to call the corresponding
    // hide_button_loading_indicator method later in the code
    // for a successful resolve/unresolve request, as that results
    // in a rerender of the message feed which replaces the button.
    buttons.show_button_loading_indicator($button);
```

Reply:
I see the general need for comments across my PR, for logic that isn't obvious on the first glance, I'll be more mindful of this going forward.

Reply:
Yes, the purpose of comments is to explain logic that isn't obvious on first glance! If you've ever read code that someone else wrote, or you wrote a while ago, it's like 3x faster to do so and make changes if it has good comments than if it doesn't.